packages/flutter_tools/lib/src/base/platform.dart

```diff
@@ -176,89 +174,119 @@ class LocalPlatform extends Platform {
-        script = platform.script,
-        executableArguments =
-            List<String>.from(platform.executableArguments),
+        _operatingSystemVersion = platform.operatingSystemVersion,
+        _localHostname = platform.localHostname,
+        _environment = Map<String, String>.from(platform.environment),
+        _executable = platform.executable,
+        _resolvedExecutable = platform.resolvedExecutable,
+        _script = platform.script,
+        _executableArguments = List<String>.from(platform.executableArguments),
         packageConfig = platform.packageConfig,
-        version = platform.version,
-        stdinSupportsAnsi = platform.stdinSupportsAnsi,
-        stdoutSupportsAnsi = platform.stdoutSupportsAnsi,
-        localeName = platform.localeName;
+        _version = platform.version,
+        _stdinSupportsAnsi = platform.stdinSupportsAnsi,
+        _stdoutSupportsAnsi = platform.stdoutSupportsAnsi,
+        _localeName = platform.localeName;
 
   @override
-  int numberOfProcessors;
+  final int numberOfProcessors;
 
   @override
-  String pathSeparator;
+  String get pathSeparator => _throwIfNull(_pathSeparator);
+  final String? _pathSeparator;
 
   @override
-  String operatingSystem;
+  final String operatingSystem;
 
   @override
-  String operatingSystemVersion;
+  String get operatingSystemVersion => _throwIfNull(_operatingSystemVersion);
+  final String? _operatingSystemVersion;
 
   @override
-  String localHostname;
+  String get localHostname => _throwIfNull(_localHostname);
+  final String? _localHostname;
 
   @override
-  Map<String, String> environment;
+  Map<String, String> get environment => _throwIfNull(_environment);
+  set environment(Map<String, String> value) {
+    _environment = value;
+  }
+  Map<String, String>? _environment;
 
   @override
-  String executable;
+  String get executable => _throwIfNull(_executable);
+  final String? _executable;
 
   @override
-  String resolvedExecutable;
+  String get resolvedExecutable => _throwIfNull(_resolvedExecutable);
+  final String? _resolvedExecutable;
 
   @override
-  Uri script;
+  Uri get script => _throwIfNull(_script);
+  final Uri? _script;
 
   @override
-  List<String> executableArguments;
+  List<String> get executableArguments => _throwIfNull(_executableArguments);
+  final List<String>? _executableArguments;
 
   @override
-  String packageConfig;
+  String? packageConfig;
 
   @override
-  String version;
+  String get version => _throwIfNull(_version);
+  final String? _version;
 
   @override
-  bool stdinSupportsAnsi;
+  bool get stdinSupportsAnsi => _throwIfNull(_stdinSupportsAnsi);
+  final bool? _stdinSupportsAnsi;
 
   @override
-  bool stdoutSupportsAnsi;
+  bool get stdoutSupportsAnsi => _throwIfNull(_stdoutSupportsAnsi);
+  final bool? _stdoutSupportsAnsi;
 
   @override
-  String localeName;
+  String get localeName => _throwIfNull(_localeName);
+  final String? _localeName;
+
+  T _throwIfNull<T>(T? value) {
```

Comment:
Maybe a TODO to remove when everything is null safe?

Reply:
Ahh thats the rub - if we want these to be non-nullable, then we probably need to provide default values for everything. That isn't how it worked before, but I don't really have a problem with making everything have a default? WDUT

Reply:
Actually, how about if we move `FakePlatform` info the test `fakes` as you tried that one other time?  `Cache.test()` can become `@required Platform platform` and we can inject a `FakePlatform` from the tests, as we already do in similar test factories with `FakeProcessManager`.  I don't love that we have a `Fake` outside the tests.